phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionObserver.java

```diff
@@ -1055,4 +1057,20 @@ public void preClose(ObserverContext<RegionCoprocessorEnvironment> c, boolean ab
     protected boolean isRegionObserverFor(Scan scan) {
         return scan.getAttribute(BaseScannerRegionObserverConstants.UNGROUPED_AGG) != null;
     }
+
+    @Override
+    public void preBatchMutate(ObserverContext<RegionCoprocessorEnvironment> c,
```

Comment:
@ritegarg @jpisaac Do we need to change the coprocessor priority for this observer ? The call order for coprocs is based on the priority.  

I see that all phoenix coprocs use QueryServicesOptions.DEFAULT_COPROCESSOR_PRIORITY. E.g. if both UngroupedAggregateRegionObserver and IndexRegionObserver both define the preBatchMutate hook, what would be the call order ?

Reply:
Looks like this is the coproc chain:
`
coprocessor$1 => '|org.apache.phoenix.coprocessor.ScanRegionObserver|805306366|', coprocessor$2 => '|org.apache.phoenix.coprocessor.UngroupedAggregateRegionObserver|805306366|', coprocessor$3 => '|org.apache.phoenix.coprocessor.GroupedAggregateRegionObserver|805306366|', coprocessor$4 => '|org.apache.phoenix.coprocessor.ServerCachingEndpointImpl|805306366|', coprocessor$5 => '|org.apache.phoenix.hbase.index.IndexRegionObserver|805306366|index.builder=org.apache.phoenix.index.PhoenixIndexBuilder,org.apache.hadoop.hbase.index.codec.class=org.apache.phoenix.index.PhoenixIndexCodec', coprocessor$6 => '|org.apache.phoenix.coprocessor.PhoenixTTLRegionObserver|805306364|'
`

Just to be sure, maybe we should change the priority?

Reply:
@ujjawal4046 @palashc Just to add to @ritegarg comments -
All mutations will be intercepted and handled thru IndexRegionObserver and henceforth IndexRegionObserver will be enabled for all Phoenix tables (being mandated as a prerequisite). 
PHOENIX-7107 will added indexes to SYSCAT, And will create a separate JIRA for enabling it on other SYSTEM tables.
Mutations on Index tables need not be blocked since those need not be replicated.
HBase tables will need to be converted to Phoenix tables, if this feature is to be effective.

Adding it to UngroupedAggregateRegionObserver in the interim.
Let us know if you see concerns.

Reply:
> Mutations on Index tables need not be blocked since those need not be replicated.

@jpisaac What do you mean by this ? The index mutations which are generated for data tables would get replicated as well.

Since it's added for IndexRegionObserver as well, I am good with the change since all coproc which have preBatchMutate are adding the logic to block mutations. @ritegarg

Reply:
@ujjawal4046 Was saying AFAIK that we do not need an explicit preBatchMutate hook for the index table regions.